packages/ui/src/views/repo/repo-branch/components/branch-list.tsx

```diff
@@ -0,0 +1,193 @@
+import { Link } from 'react-router-dom'
+
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+  Badge,
+  Button,
+  CommitCopyActions,
+  CopyButton,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+  DropdownMenuShortcut,
+  DropdownMenuTrigger,
+  Icon,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Text
+} from '@/components'
+import { getInitials } from '@utils/stringUtils'
+
+import { DivergenceGauge } from './divergence-gauge'
+
+interface BranchProps {
+  id: number
+  name: string
+  sha: string
+  timestamp: string
+  user: {
+    name: string
+    avatarUrl?: string
+  }
+  checks?: {
+    done?: number
+    total?: number
+    status?: number
+  }
+  behindAhead: {
+    behind?: number
+    ahead?: number
+    default?: boolean
+  }
+}
+
+interface PageProps {
+  branches: BranchProps[]
+  spaceId?: string
+  repoId?: string
+  defaultBranch?: string
+}
+
+export const BranchesList = ({ branches, spaceId, repoId, defaultBranch }: PageProps) => {
+  const moreActionsTooltip = (branchInfo: BranchProps) => {
```

Comment:
this should be a separate component outside BranchesList. otherwise it'll get recreated everytime the parent re-renders